test/lit/wat-kitchen-sink.wast

```diff
@@ -25,6 +25,12 @@
 
```

Comment:
Could we add a text roundtrip test in this file perhaps? (to also test the nested versions in the new parser)

Reply:
The new parser does not support the non-standard `func_subtype`, `struct_subtype`, and `array_subtype` type definition format that we emit by default, so it can't currently round trip the file. My plan was to make our printer emit the standard format just before switching the default parser over, but it's probably worth doing that sooner rather than later.